src/lib/markdown_generators.js

```diff
@@ -0,0 +1,182 @@
+    course_info:        {
+      instructors: courseData["instructors"].map(
+        instructor =>
+          `Prof. ${instructor["first_name"]} ${instructor["last_name"]}`
+      ),
+      department: titleCase.titleCase(
+        courseData["url"].split("/")[2].replace(/-/g, " ")
+      ),
+      topics:        courseData["course_collections"].map(makeTopic),
+      course_number: courseNumber,
+      term:          `${courseData["from_semester"]} ${courseData["from_year"]}`,
+      level:         courseData["course_level"],
+      menu:          {
+        main: {
+          weight: -10
+        }
+      }
+    }
+  }
+  return `---\n${yaml.safeDump(frontMatter)}---\n`
+}
+
+const generateCourseSectionFrontMatter = (title, menuIndex) => {
+  /*
+      Generate the front matter metadata for a course section given a title and menu index
+      */
+  return `---\n${yaml.safeDump({
+    title: title,
+    menu:  {
+      main: {
+        weight: menuIndex
+      }
+    }
+  })}---\n`
+}
+
+const generateCourseFeatures = courseData => {
+  /*
+      Generate markdown for the "Course Features" section of the home page
+      */
+  let courseFeaturesMarkdown = markdown.headers.hX(5, "Course Features")
+  const courseFeatures = []
+  courseData["course_features"].forEach(courseFeature => {
+    const url = courseFeature["ocw_feature_url"]
+      .replace("/index.htm", "/")
+      .split("/")[-1]
+    courseFeatures.push(markdown.misc.link(courseFeature["ocw_feature"], url))
+  })
+  courseFeaturesMarkdown += `\n${markdown.lists.ul(courseFeatures)}`
+  return courseFeaturesMarkdown
+}
+
+const generateCourseCollections = courseData => {
+  /*
+      Generate markdown for the "Course Collections" section of the home page
+      */
+  let courseCollectionsMarkdown = markdown.headers.hX(5, "Course Collections")
+  courseCollectionsMarkdown +=
+    "\nSee related courses in the following collections:\n"
+  courseCollectionsMarkdown += `\n${markdown.emphasis.i(
+    "Find Courses by Topic"
+  )}\n\n`
+  const courseCollections = []
+  courseData["course_collections"].forEach(courseCollection => {
+    const feature = courseCollection["ocw_feature"]
+    const subfeature = courseCollection["ocw_subfeature"]
+    const specialty = courseCollection["ocw_specialty"]
+    let collection = feature
+    if (subfeature) {
+      collection += ` > ${subfeature}`
+    }
+    if (specialty) {
+      collection += ` > ${specialty}`
+    }
+    courseCollections.push(markdown.misc.link(collection, "#"))
+  })
+  courseCollectionsMarkdown += markdown.lists.ul(courseCollections)
+  return courseCollectionsMarkdown
+}
+
+const generateCourseSectionMarkdown = (page, courseData) => {
+  /*
+      Generate markdown a given course section page
+      */
+  let htmlStr = page["text"]
+  courseData["course_pages"].forEach(coursePage => {
+    const placeholder = `./resolveuid/${coursePage["uid"]}`
+    const pageName = coursePage["short_url"].replace(".html", "")
```

Comment:
What is the `replace` doing here?

Reply:
This is an artifact from the code I ported from ocw-data-parser.  The function this is used in there looks at all the pages, including the index, which has a ".html" in the short_url.  It's not necessary here so I'll remove it.